File: java/src/main/java/io/epoch/AeronTransport.java
package io.epoch;

import java.util.List;

public final class AeronTransport implements Transport {
    public record AeronConfig(String channel, int streamId, String aeronDirectory) {
    }

    private final AeronConfig config;

    public AeronTransport(AeronConfig config) {
        this.config = config;
    }

    @Override
    public void send(Engine.Message message) {
        throw new UnsupportedOperationException("Aeron transport not linked");
    }

    @Override
    public List<Engine.Message> poll(int max) {
        throw new UnsupportedOperationException("Aeron transport not linked");
    }

    @Override
    public void close() {
        // no-op for stub
    }

    public AeronConfig config() {
        return config;
    }
}

File: python/epoch/aeron_transport.py
from __future__ import annotations

from dataclasses import dataclass
from typing import List

from .engine import Message
from .transport import Transport


@dataclass(frozen=True)
class AeronConfig:
    channel: str
    stream_id: int
    aeron_directory: str


class AeronTransport(Transport):
    def __init__(self, config: AeronConfig) -> None:
        self._config = config

    @property
    def config(self) -> AeronConfig:
        return self._config

    def send(self, message: Message) -> None:
        raise NotImplementedError("Aeron transport not linked")

    def poll(self, max_items: int) -> List[Message]:
        raise NotImplementedError("Aeron transport not linked")

    def close(self) -> None:
        return None
